src/directives/routeClick.js

```diff
@@ -0,0 +1,41 @@
+function routeClickFactory (Route, $location, $window, $timeout) {
+  'ngInject'
+
+  return {
+    restrict: 'A',
+    scope: true,
+
+    link (scope, element, attrs) {
+      element.mouseup((event) => {
+        event.preventDefault();
+        let urlPath = element.attr('href');
+
+        if (event.which === 2 || (event.which === 1 && (event.metaKey || event.ctrlKey))) {
+          const fullUrl = $window.location.origin + '/' + urlPath;
+          return $window.open(fullUrl, '_blank');
+        } else {
+          if (!Route.isHtml5ModeEnabled()) {
+            urlPath = urlPath.replace(/^#/, '');
+          }
+          return $timeout(() => $location.url(urlPath));
+        }
+      });
+
+      const object = Route.getUrlWriters();
+      for (const writerName in object) {
+        const writer = object[writerName];
+        scope[`${writerName}UrlWriter`] = writer;
+      }
+
+      return scope.$watch(attrs.routeClick, (newUrl) => {
```

Comment:
yeah i agree with consolidation i think there is some overlap.

few comments:
- route-on-click sounds good, closer to the on-click
- can route-href still be an alias of route-on-click (so we don't have to update our markup?)
- control key (and meta) should also be a constant?
- are we 100% sure we want to consolidate into 1 directive?

Reply:
I'd like route-href to go away - part of the point of my suggestion is that we then only have one directive people have to remember to use, so if the code is littered with route-href people will just keep using it.

ctrl / meta key don't need explicit constants because they're named ... the buttons are otherwise numbers that make the code difficult to understand

I think the consolidation is a good idea (otherwise I wouldn't have suggested it) but I'm open to discussion on it. Do you have concerns, if so, what are they? Perhaps Kyle might want to chime in as well.

Reply:
Cool yeah i've started upgrading the directive i like only having 1 so route-href can die! 💯 

Is it ok that the current route-href links open now on the 'mouseup' event?